frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Select, SelectVariant, SelectOption, SelectGroup } from '@patternfly/react-core';
+import { ShowFiltersKeyValue, ExpandFiltersKeyValue, DisplayFilters } from './filter-utils';
+import './FilterDropdown.scss';
+
+type FilterDropdownProps = {
+  filters: DisplayFilters;
+  onChange: (filter: DisplayFilters) => void;
+};
+
+const FilterDropdown: React.FC<FilterDropdownProps> = ({ filters, onChange }) => {
+  const [isOpen, setIsOpen] = React.useState();
+  const [selected, setSelected] = React.useState<string[]>(
+    Object.keys(filters)
+      .filter((key) => filters[key])
+      .map((key) => ShowFiltersKeyValue[key] || ExpandFiltersKeyValue[key]),
+  );
```

Comment:
The filters are stored in redux. Why is there local state for the same data? It should be a controlled component using the prop values.

Reply:
So Redux is connected to the Toolbar and when dropdown is open and one of the options is selected, it is stored in the store the moment it is selected but it's not reflected in the dropdown until it is closed and opened again. That's why I used a local state.

Reply:
If we can store it in the store, then we can store it in redux and when that happens it should result in a re-render of this dropdown component.
That's standard redux. We shouldn't need local state.

Reply:
fixed